perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -979,4 +979,21 @@ private String getGroupStatusSQLConditionForMembersPage(MembersPageQuery query,
 		}
 		return groupStatusesQueryString;
 	}
+
+	public void rejectGroupApplications(PerunSession session, Member member) {
+		try {
+			List<Integer> ids = jdbc.query(
+				"select id from application " +
+					"where user_id=? and vo_id=? and group_id is not null and state not in (?)",
+				new SingleColumnRowMapper<>(Integer.class), member.getUserId(), member.getVoId(), rejected);
+
+			// Reject all applications for member
+			for (Integer id : ids) {
```

Comment:
Just wondering if this should not be done directly in DB instead of looping through IDs. And actually, I believe it could be then merged into a single query.
```
UPDATE application SET state = ?, modified_at =?, modified_by_uid=?,
WHERE id IN
( SELECT id FROM application WHERE user_id = ? AND vo_id = ? AND group_id IS NOT NULL AND state != ?)
```
@xflord, @HejdaJakub any opinions?

Reply:
Yes, this could be done. I'll try to implement this :+1: